Sources/GRPCCore/Timeout.swift

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright 2023, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import Dispatch
+
+/// A timeout for a gRPC call.
+///
+/// Timeouts must be positive and at most 8-digits long.
+/// See "Timeout" in https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md#requests
+@available(macOS 13.0, iOS 16.0, watchOS 9.0, tvOS 16.0, *)
+public struct Timeout: CustomStringConvertible, Equatable {
+  /// The largest amount of any unit of time which may be represented by a gRPC timeout.
+  internal static let maxAmount: Int64 = 99_999_999
+
+  /// The wire encoding of this timeout as described in the gRPC protocol.
+  /// See: https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md.
+  public let wireEncoding: String
+  public let duration: Duration
+
+  public var description: String {
+    return self.wireEncoding
+  }
+
+  public init?(stringLiteral value: String) {
+    guard 2 ... 8 ~= value.count else {
```

Comment:
I would recommend using pattern matching instead of calling the pattern matching operator directly
```suggestion
    guard case 2...8 = value.count else {
```

Reply:
Is this for readability or is there some other benefit to doing this?

Reply:
@gjcairo just for readability. 
@glbrntt fine with this as well.